src/main/thrift/parquet.thrift

```diff
@@ -877,3 +896,31 @@ struct FileMetaData {
   7: optional list<ColumnOrder> column_orders;
 }
 
+/**
+ * Supported encryption algorithms.
+ */
+enum EncryptionAlgorithm {
```

Comment:
Just some thoughts: I'm not sure how likely it is that we'd like to use additional encryption algorithms in addition to AES in the future, but instead of enum maybe we should consider using structs with algorithm specific parameters as fields in the corresponding struct (like for example iv_prefix, for AES struct, which is now stored for every encryption algorithm, even when it is not used) similarly like in the new LogicalType API. It would provide more flexible API wouldn't it?

Reply:
Even if we would never need parameters for the enryption algorithms, enum-s are problematic because extending an enum is a breaking change in thrift, so if this list ever grows, older readers may have problem reading files with newer encryption algortithms. Of course, they wouldn't be able to read the encrypted data anyway, since they don't have the implementation, but at least they could read the rest of the file.

The best way we found to deal with this problem is using a union of structs instead of enums. This has the additional advantage that some of these structs can have extra fields, thereby we can have "rich" enums. Please see LogicalType in parquet.thrift.

Reply:
@nandorKollar , sorry for a late response, I missed the github notification in the mail stream..
I'm fine with your and @zivanfi suggestion of using struct / union for this. While IVs are used in AES GCM and CTR flavors, there are ciphers working without an IV. And having forward compatibility is obviously important. I'll update the code.

Reply:
@nandorKollar Looks like practically  there is no chance we'll use a cipher without an IV in the future, so I'm leaving the iv_prefix in the FileCryptoMetaData. It was defined as optional in any case, because its a prefix, not the full iv. 
In parallel, I'm adding an "aad_metadata" field to the cipher-specific structures that work with AADs (used in AES-GCM). It might be useful to retrieve the AAD during file reading.